include/host1x.h

```diff
@@ -66,6 +66,55 @@ struct host1x_gr3d;
 struct host1x_bo;
 struct host1x;
 
+#define PIX_BUF_FMT(id, bpp, tiled) \
+	((tiled) << 16 | (id) << 8 | (bpp))
+
+#define PIX_BUF_FORMAT(f) \
+	((f) & 0xffff)
+
+#define PIX_BUF_FORMAT_BITS(f) \
+	((f) & 0xff)
+
+#define PIX_BUF_FORMAT_BYTES(f) \
+	(PIX_BUF_FORMAT_BITS(f) >> 3)
+
+#define PIX_BUF_FORMAT_TILED(f) \
+	(((f) >> 16) & 1)
+
+enum pixel_format {
+    PIX_BUF_FMT_A8            = PIX_BUF_FMT(0, 8, 0),
+    PIX_BUF_FMT_L8            = PIX_BUF_FMT(1, 8, 0),
+    PIX_BUF_FMT_S8            = PIX_BUF_FMT(2, 8, 0),
+    PIX_BUF_FMT_LA88          = PIX_BUF_FMT(3, 16, 0),
+    PIX_BUF_FMT_RGB565        = PIX_BUF_FMT(4, 16, 0),
+    PIX_BUF_FMT_RGBA5551      = PIX_BUF_FMT(5, 16, 0),
+    PIX_BUF_FMT_RGBA4444      = PIX_BUF_FMT(6, 16, 0),
+    PIX_BUF_FMT_D16_LINEAR    = PIX_BUF_FMT(7, 16, 0),
+    PIX_BUF_FMT_D16_NONLINEAR = PIX_BUF_FMT(8, 16, 0),
+    PIX_BUF_FMT_RGBA8888      = PIX_BUF_FMT(9, 32, 0),
+    PIX_BUF_FMT_RGBA_FP32     = PIX_BUF_FMT(10, 32, 0),
+
+    PIX_BUF_FMT_A8_TILED            = PIX_BUF_FMT(0, 8, 1),
+    PIX_BUF_FMT_L8_TILED            = PIX_BUF_FMT(1, 8, 1),
+    PIX_BUF_FMT_S8_TILED            = PIX_BUF_FMT(2, 8, 1),
+    PIX_BUF_FMT_LA88_TILED          = PIX_BUF_FMT(3, 16, 1),
+    PIX_BUF_FMT_RGB565_TILED        = PIX_BUF_FMT(4, 16, 1),
+    PIX_BUF_FMT_RGBA5551_TILED      = PIX_BUF_FMT(5, 16, 1),
+    PIX_BUF_FMT_RGBA4444_TILED      = PIX_BUF_FMT(6, 16, 1),
+    PIX_BUF_FMT_D16_LINEAR_TILED    = PIX_BUF_FMT(7, 16, 1),
+    PIX_BUF_FMT_D16_NONLINEAR_TILED = PIX_BUF_FMT(8, 16, 1),
+    PIX_BUF_FMT_RGBA8888_TILED      = PIX_BUF_FMT(9, 32, 1),
+    PIX_BUF_FMT_RGBA_FP32_TILED     = PIX_BUF_FMT(10, 32, 1),
```

Comment:
Hmm, I'm not loving mixing the tiling into the pixel format, as those are completely orthogonal features... Tiling is the order the pixels are stored, whereas the pixel format is the order the components of the pixels are stored. IIRC, the hardware even tiles on byte-sizes, as opposed to pixel-sizes.

I think we should instead have an `enum pixel_layout { PIX_BUF_LAYOUT_LINEAR = 0, PIX_BUF_LAYOUT_TILED = 1 }` or something.